### PR TITLE
Implement HTTP-based cache serializer-deserializer.

### PR DESCRIPTION
**Pull request by scottgifford**
https://issues.apache.org/jira/projects/HTTPCORE/issues/HTTPCORE-615

### CODE REVIEW

File: httpclient-cache/src/test/java/org/apache/http/impl/client/cache/BenchmarkMemcachedCacheEntryHttp.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+     * @throws Exception if anything goes wrong
+     */
+    private static void benchmarkSerializeDeserialize(final String testName, final String storageKey, final HttpCacheEntry testEntry, final MemcachedCacheEntryFactory factory) throws Exception {
+        final MemcachedCacheEntry memcachedCacheEntry = factory.getMemcachedCacheEntry(storageKey, testEntry);
+        final byte[] testBytes = memcachedCacheEntry.toByteArray();
+
+        // Verify once to make sure everything is right, and maybe warm things up a little
+        verifyHttpCacheEntryFromBytes(storageKey, testEntry, factory, testBytes);
+
+        System.out.printf("%40s: %6d bytes\n", testName + " serialized size", testBytes.length);
```
Comment (michael-o) [inline httpclient-cache/src/test/java/org/apache/http/impl/client/cache/BenchmarkMemcachedCacheEntryHttp.java]: `\n` > `%n`

File: httpclient-cache/src/test/java/org/apache/http/impl/client/cache/BenchmarkMemcachedCacheEntryHttp.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+        final long start = System.nanoTime();
+        final int runs = 5000;
+        for(int i=0;i<runs;i++) {
+            func.run();
+        }
+        final long end = System.nanoTime();
+        final long elapsed = end - start;
+        final long each = elapsed / runs;
+        System.out.printf("%40s: %6d runs in %7.3f ms, %7.3f ms/run\n", testName, runs, elapsed/1000000.0, each/1000000.0);
```
Comment (michael-o) [inline httpclient-cache/src/test/java/org/apache/http/impl/client/cache/BenchmarkMemcachedCacheEntryHttp.java]: Same here.

Also use TimeUnit..toMilli.. it makes the code more readable.

File: httpclient-cache/src/test/java/org/apache/http/impl/client/cache/MemcachedCacheEntryHttpTestUtils.java [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+
+    /**
+     * Create a new HttpCacheEntry object with default fields, except for those overridden by name in the template.
+     *
+     * @param template Map of field names to override in the test object
+     * @return New object for testing
+     */
+    static HttpCacheEntry buildSimpleTestObjectFromTemplate(final Map<String, Object> template) {
+        final Resource resource = getOrDefault(template, "resource",
+                new HeapResource("Hello World".getBytes(Charset.forName("UTF-8"))));
```
Comment (michael-o) [inline httpclient-cache/src/test/java/org/apache/http/impl/client/cache/MemcachedCacheEntryHttpTestUtils.java]: Use `StandardCharsets`.

**Comment by ok2c** [comment]
Superseded by #192

**Pull request closed without merging**